app/services/enrollment_email_templates_retriever.rb

```diff
@@ -0,0 +1,97 @@
+class EnrollmentEmailTemplatesRetriever
+  attr_reader :enrollment,
+              :instructor
+
+  def initialize(enrollment, instructor)
+    @enrollment = enrollment
+    @instructor = instructor
+  end
+
+  def perform
+    email_kinds.map do |email_kind|
+      build_template(email_kind)
+    end
+  end
+
+  private
+
+  def build_template(email_kind)
+    EnrollmentEmailTemplate.new(
+      action_name: "#{email_kind}_application",
+      content_plain_text: render_template(email_kind),
+    )
+  end
+
+  def render_template(email_kind)
+    if custom_template_exists?(email_kind)
+      render_specific_template_without_layout(email_kind)
+    else
+      render_default_template_with_layout(email_kind)
+    end
+  end
+
+  def render_specific_template_without_layout(email_kind)
+    renderer.render(
+      file: "enrollment_mailer/#{enrollment.target_api}/#{email_kind}_application",
+      layout: false,
+    )
+  end
+
+  def render_default_template_with_layout(email_kind)
+    renderer.render(
+      file: "enrollment_mailer/#{email_kind}_application",
+      layout: 'layouts/enrollment_mailer',
+    )
+  end
+
+  def custom_template_exists?(email_kind)
+    File.exists?(
+      Rails.root.join(
+        "app/views/enrollment_mailer/#{enrollment.target_api}/#{email_kind}_application.text.erb"
+      )
+    )
+  end
+
+  def renderer
+    @renderer ||= ActionView::Base.new('app/views', variables)
+  end
+
+  def variables
+    @variables ||= {
+      url:              enrollment_url,
+      target_api_label: target_api_label,
+      front_url:        front_url,
+      user:             user,
+      enrollment:       enrollment,
+      instructor:       instructor,
+    }
+  end
+
+  def enrollment_url
+    "#{front_url}/#{enrollment.target_api.tr('_', '-')}/#{enrollment.id}"
+  end
+
+  def target_api_label
+    target_api_data['target_api']
+  end
+
+  def front_url
+    ENV.fetch("FRONT_HOST")
+  end
+
+  def user
+    @user ||= enrollment.user
+  end
+
+  def email_kinds
+    %w[
+      refuse
+      review
+      validate
```

Comment:
I just noticed that the `notify` action is missing here. This action also require a template to be send in the front.

Reply:
What's the difference with `review` here ?

I'll add it.

Reply:
Btw is it `notify` or `notify_application_sent` ?

I'm not sure so I added `notify` here https://github.com/betagouv/signup-back/pull/200/commits/ed42655bff521c52a6e0ba206c57492ad1ec1607

Reply:
`notify_application_sent` is the mail send to instructors when a user submit a new authorization request

`notify` is the mail send when an instructor wants to send a message to the owner of the authorization request.

Only the second one need a template.

Reply:
Ok thanks, I mark this thread as resolved thanks to https://github.com/betagouv/signup-back/commit/ed42655bff521c52a6e0ba206c57492ad1ec1607